### PR TITLE
[14.0] [mig] hr_ timesheet_sheet

### PR DESCRIPTION
**Pull request by schhatbar-initos**

### CODE REVIEW

File: hr_timesheet_sheet/models/hr_timesheet_sheet.py [hunk 1]
```diff
@@ -0,0 +1,897 @@
...
+        return self.env["hr.employee"].search(
+            [("user_id", "=", self.env.uid), ("company_id", "in", [company.id, False])],
+            limit=1,
+            order="company_id ASC",
+        )
+
+    def _default_department_id(self):
+        return self._default_employee().department_id
+
+    name = fields.Char(compute="_compute_name", context_dependent=True)
```
Comment (CasVissers-360ERP) [inline hr_timesheet_sheet/models/hr_timesheet_sheet.py]: > context_dependent', if this is an actual parameter you may want to override the method _valid_field_parameter on the relevant model in order to allow it

**Review by CasVissers-360ERP — CHANGES_REQUESTED** [review]
Initial feedback

**Review by fshah-initos — APPROVED** [review]
LGTM

**Review by fshah-initos — APPROVED** [review]

**Comment by schhatbar-initos** [comment]
@CasVissers-360ERP Can you please review this PR? 
Thanks in advance!

**Review by CasVissers-360ERP — APPROVED** [review]

**Comment by schhatbar-initos** [comment]
@alexey-pelykh Can you please merge this PR hence it is approved?
Thanks in Advance!

**Comment by schhatbar-initos** [comment]
@pedrobaeza  Can you please merge this PR hence it is approved? This module(hr_timesheet_sheet) is also dependent on some other modules.
Thanks in Advance!

File: hr_timesheet_sheet/models/hr_timesheet_sheet.py [hunk 2]
```diff
@@ -0,0 +1,897 @@
...
+
+    def _get_possible_reviewers(self):
+        self.ensure_one()
+        res = self.env["res.users"].browse(SUPERUSER_ID)
+        if self.review_policy == "hr":
+            res |= self.env.ref("hr.group_hr_user").users
+        elif self.review_policy == "hr_manager":
+            res |= self.env.ref("hr.group_hr_manager").users
+        elif self.review_policy == "timesheet_manager":
+            res |= self.env.ref("hr_timesheet.group_timesheet_manager").users
```
Comment (CasVissers-360ERP) [inline hr_timesheet_sheet/models/hr_timesheet_sheet.py]: Should be group_hr_timesheet_approver not timesheet_manager?

File: hr_timesheet_sheet/models/hr_timesheet_sheet.py [hunk 3]
```diff
@@ -0,0 +1,897 @@
...
+        self.ensure_one()
+        subscribers = self._get_possible_reviewers().mapped("partner_id")
+        subscribers |= self._get_informables()
+        return subscribers
+
+    def _timesheet_subscribe_users(self):
+        for sheet in self.sudo():
+            subscribers = sheet._get_subscribers()
+            if subscribers:
+                self.message_subscribe(partner_ids=subscribers.ids)
```
Comment (CasVissers-360ERP) [inline hr_timesheet_sheet/models/hr_timesheet_sheet.py]: self.message_subscribe(partner_ids=subscribers.ids) should be sheet.message_subscribe(partner_ids=subscribers.ids)

**Comment by schhatbar-initos** [comment]
@CasVissers-360ERP  The suggested changes has been done.
Can you please review this PR?
Need to get it merged asap because this module is dependent upon many other modules.
Thanks in advance!

**Comment by schhatbar-initos** [comment]
@pedrobaeza Can you please check with this PR?

File: hr_timesheet_sheet/security/ir.model.access.csv [hunk 4]
```diff
@@ -1,2 +1,4 @@
 id,name,model_id:id,group_id:id,perm_read,perm_write,perm_create,perm_unlink
 access_hr_timesheet_sheet_user,hr_timesheet.sheet,model_hr_timesheet_sheet,base.group_user,1,1,1,1
+access_hr_timesheet_sheet_line,hr_timesheet.sheet.line,model_hr_timesheet_sheet_line,base.group_user,1,1,1,1
```
Comment (pedrobaeza) [inline hr_timesheet_sheet/security/ir.model.access.csv]: Shouldn't this be to the user with timesheet permissions only?

Comment (pedrobaeza) [reply hr_timesheet_sheet/security/ir.model.access.csv]: The following line is the same case.

**Pull request merged**